fabfile.py

```diff
@@ -149,57 +130,77 @@ def curl(c):
   raise BaseException("Curl could not be installed")
 
 
-def provision_tasks(c, host, name):
-  fullname = get_fullname(name)
+@task
+def add_application(ctx, application):
+  c = Connection(host=ctx.config.get('host'), user=USER)
+  c.config = ctx.config
+
+  app = get_application(c, application)
+  if not app:
+    print("Problème de nom avec %s" % application)
+    return
+
+  setup_application(c, app)
+  refresh_tasks(c, force=True, application=app)
+
+
+def provision_tasks(c):
+  fullname = c.config.get('fullname')
 
   system(c, fullname)
   nginx_setup(c)
   node(c)
   mongodb(c)
 
-  monitor(c)
-
-  python(c)
-  openfisca_setup(c)
-  openfisca_config(c)
-
-  app_setup(c)
+  # monitor(c) # TODO
 
   letsencrypt(c)
-  nginx_all_sites(c, fullname)
+  nginx_all_sites(c)
 
-  refresh_tasks(c, force=True)
+  for application in c.config.applications:
+    setup_application(c, application)
 
+  refresh_tasks(c, force=True)
 
-def get_fullname(name):
-  return "%s.mes-aides.1jeune1solution.beta.gouv.fr" % name
 
+def setup_application(c, application):
+  node_setup(c, application)
+  openfisca_setup(c, application)
+  generate_deploy_key(c, application)
 
-@task
-def add_next(ctx, host):
-  c = Connection(host=host, user=USER)
-  # app_setup(c, ANGULAR_FOLDER, 'angular')
-  # app_refresh(c, ANGULAR_FOLDER)
 
+def print_dns_records(config):
+  fullname = config.get('fullname')
+  dns_root = config.get('dns_root')
+  dns_fullname = fullname[0:len(fullname)-len(dns_root)].strip(".")
```

Comment:
```suggestion
  dns_fullname = fullname[:-len(f".{dns_root}")]
```
ou
dns_fullname = fullname[:-len(dns_root)].rstrip(".")